apis/medusa/v1alpha1/medusa_types.go

```diff
@@ -0,0 +1,148 @@
+
+	// Maximum number of backups to keep (used by the purge process).
+	// Default is unlimited.
+	// +optional
+	MaxBackupCount int `json:"maxBackupCount,omitempty"`
+
+	// AWS Profile to use for authentication.
+	// +optional
+	ApiProfile string `json:"apiProfile,omitempty"`
+
+	// Max upload bandwidth in MB/s.
+	// Defaults to 50 MB/s.
+	// +optional
+	TransferMaxBandwidth int `json:"transferMaxBandwidth,omitempty"`
+
+	// Number of concurrent uploads.
+	// Helps maximizing the speed of uploads but puts more pressure on the network.
+	// Defaults to 1.
+	// +optional
+	ConcurrentTransfers int `json:"concurrentTransfers,omitempty"`
+
+	// File size over which cloud specific cli tools are used for transfer.
+	// Defaults to 100 MB.
+	// +optional
+	MultiPartUploadThreshold int `json:"multiPartUploadThreshold,omitempty"`
+
+	// Host to connect to for the storage backend.
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// Region of the storage bucket.
+	// Defaults to "default".
+	// +optional
+	Region string `json:"region,omitempty"`
+
+	// Port to connect to for the storage backend.
+	Port int `json:"port,omitempty"`
+
+	// Whether to use SSL for the storage backend.
+	Secure bool `json:"secure,omitempty"`
+
+	// Age after which orphan sstables can be deleted from the storage backend.
+	// Protects from race conditions between purge and ongoing backups.
+	// Defaults to 10 days.
+	// +optional
+	BackupGracePeriodInDays int `json:"backupGracePeriodInDays,omitempty"`
+}
+
+type ContainerImage struct {
+
+	// +kubebuilder:default="docker.io"
+	// +optional
+	Registry *string `json:"registry,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Repository string `json:"repository"`
+
+	// +kubebuilder:default="latest"
+	// +optional
+	Tag *string `json:"tag,omitempty"`
+
+	// +kubebuilder:default="IfNotPresent"
+	// +optional
+	PullPolicy *corev1.PullPolicy `json:"pullPolicy,omitempty"`
+
+	// +optional
+	PullSecret corev1.LocalObjectReference `json:"imagePullSecret,omitempty"`
+}
+
+type MedusaClusterTemplate struct {
+	// MedusaContainerImage is the image characteristics to use for Medusa containers. Leave nil
+	// to use a default image.
+	// +optional
+	Image ContainerImage `json:"medusaContainerImage,omitempty"`
+
+	// SecurityContext applied to the Medusa containers.
+	// +optional
+	SecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
+
+	// Defines the username and password that Medusa will use to authenticate CQL connections to Cassandra clusters.
+	// These credentials will be automatically turned into CQL roles by cass-operator when bootstrapping the datacenter,
+	// then passed to the Medusa instances, so that it can authenticate against nodes in the datacenter using CQL.
+	// The secret must be in the same namespace as Reaper itself and must contain two keys: "username" and "password".
+	// +optional
+	CassandraUserSecretRef string `json:"cassandraUserSecretRef,omitempty"`
```

Comment:
Prefer `corev1.LocalObjectReference` when referencing other objects in the same namespace.